ark/utils/test_utils.py

```diff
@@ -0,0 +1,168 @@
+        tif_data = tif_data.reshape((fov_number, *img_shape, chan_number)).astype(dtype)
+    else:
+        tif_data = np.full((*img_shape, fov_number, chan_number),
+                           list(fills.values()), dtype=dtype)
+        tif_data = np.moveaxis(tif_data, 2, 0)
+
+    return tif_data
+
+
+def _create_tifs(base_dir, fov_names, img_names, shape, sub_dir, fills, dtype):
+    tif_data = _gen_tif_data(len(fov_names), len(img_names), shape, fills, dtype)
+
+    if sub_dir is None:
+        sub_dir = ""
+
+    filelocs = {}
+
+    for i, fov in enumerate(fov_names):
+        filelocs[fov] = []
+        fov_path = os.path.join(base_dir, fov, sub_dir)
+        os.makedirs(fov_path)
+        for j, name in enumerate(img_names):
+            io.imsave(os.path.join(base_dir, name), tif_data[i, :, :, j])
+            filelocs[fov].append(os.path.join(base_dir, fov, sub_dir, name))
+
+    return filelocs, tif_data
+
+
+def _create_multitiff(base_dir, fov_names, channel_names, shape, sub_dir, fills, dtype):
+    tif_data = _gen_tif_data(len(fov_names), len(channel_names), shape, fills, dtype)
+
+    filelocs = {}
+
+    for i, fov in enumerate(fov_names):
+        tiffpath = os.path.join(base_dir, f'{fov}.tiff')
+        io.imsave(tiffpath, tif_data[i, :, :, :], plugin='tifffile')
+        filelocs[fov] = tiffpath
+
+    return filelocs, tif_data
+
+
+def _create_mibitiff(base_dir, fov_names, channel_names, shape, sub_dir, fills, dtype):
+    tif_data = _gen_tif_data(len(fov_names), len(channel_names), shape, fills, dtype)
+
+    filelocs = {}
+
+    mass_map = tuple(enumerate(channel_names, 1))
+
+    for i, fov in enumerate(fov_names):
+        tif_obj = mi.MibiImage(tif_data[i, :, :, :],
+                               mass_map,
+                               **METADATA)
+
+        tiffpath = os.path.join(base_dir, f'{fov}.tiff')
+        tiff.write(tiffpath, tif_obj, dtype=dtype)
+        filelocs[fov] = tiffpath
+
+    return filelocs, tif_data
+
+
+TIFFMAKERS = {
+    'tiff': _create_tifs,
+    'multitiff': _create_multitiff,
+    'mibitiff': _create_mibitiff,
+}
+
+
+def create_paired_xarray_fovs(base_dir, fov_names, channel_names, img_shape=(1024, 1024),
+                              mode='tiff', delimiter=None, sub_dir=None, fills=None, dtype="int8"):
+
+    if not os.path.isdir(base_dir):
+        raise FileNotFoundError(f'{base_dir} is not a directory')
+        return None
+
+    if fov_names is None or fov_names is []:
+        raise ValueError('No fov names were given...')
+        return None
+
+    if channel_names is None or channel_names is []:
+        raise ValueError('No image names were given...')
+        return None
+
+    if not isinstance(fov_names, list):
+        fov_names = [fov_names]
+
+    if not isinstance(channel_names, list):
+        channel_names = [channel_names]
+
+    # verify fills structure -> dict(fov_name : list [num_channels])
+    if fills is not None:
```

Comment:
Which test functions currently require specified values in the input data?

Reply:
I'm not sure if any do atm, but I figured having distinct 'channel values' would be a good check that things are loaded in the correct order (w.r.t. both fovs and channels).  We weren't doing this before, but I think we probs should in case anyone breaks things.

That being said, I'm open to not doing that, or to doing it differently; I recognize that the fills structure is kinda weird.

Reply:
btw, the reason we weren't doing this earlier was because we didn't have a ground-truth xarray to compare the values to, since we were mostly just checking for correct xarray structure on loaded example images.

Reply:
I agree, I think it's a good idea, and most of the deepcell tests I've written do this. Given how flexible the overall function is, it just makes this parameter a bit trickier to implement

Reply:
The other approach I see accomplishing the same thing with less explicit structure needed is having `fills=False` randomize data vals, and `fills=True` mean 'set-order' filling, i.e:

fov1: chan1 is filled with 0's, chan2 is filled with 1's, ..., chanN is filled with (N-1)'s
fov2: chan1 is filled with N's, ...

and then just have any values, K>256, wrap around back into [0, 255].

This is probably better since the user doesn't have to explicitly write the exotic structure of `fills`, but lmk what you think.

Reply:
I agree, I think that's a great solution. Especially for channel data (not label data), I think the need to have super fine-grained control over the specific labels is fairly limited.